src/read/read_cache.rs

```diff
@@ -129,6 +129,10 @@ impl<'a, R: Read + Seek> ReadRef<'a> for &'a ReadCache<R> {
         // This is OK because we never mutate or remove entries.
         Ok(unsafe { mem::transmute::<&[u8], &[u8]>(buf) })
     }
+
+    fn read_bytes_at_until_sequence(self, offset: usize, needle: &[u8], max_end: usize) -> Result<&'a [u8], ()> {
```

Comment:
There are two missing implementations here (which I did not need to correctly test this rich header parsing)

Before digging into what a read cache is, I'd like to have your opinion on my pull request. Then, if you're OK with it, I'll try to implement `read_bytes_at_until_sequence` here

Reply:
OK, it's not needed anymore